src/lib/comms/sol-lwm2m.c

```diff
@@ -4830,6 +5150,14 @@ sol_lwm2m_client_start(struct sol_lwm2m_client *client)
 
     return 0;
 
+err_access_control:
+    if (client->supports_access_control) {
+        ctx = find_object_ctx_by_id(client, ACCESS_CONTROL_OBJECT_ID);
+        while ((instance = sol_vector_get(&ctx->instances, 0)) != NULL) {
```

Comment:
Because `sol_vector_del_element(&ctx->instances, instance);`.

Since I'm deleting stuff, it messes up the counter in SOL_VECTOR_FOREACH_IDX... same thing as https://github.com/solettaproject/soletta/pull/2255/files#diff-e4058dc14a343a8aceb7210b323358a2R3165, which is where I first faced this.